src/kv/deserialise.h

```diff
@@ -78,6 +80,11 @@ namespace kv
       throw std::logic_error("get_index not implemented");
     }
 
+    aft::Request& get_request() override
+    {
+      throw std::logic_error("get_index not implemented");
```

Comment:
```suggestion
      throw std::logic_error("get_request not implemented");
```

Could async execution be implemented as a Mix-in? It might be easier to test too.